### ISSUE TITLE
[BUG] iOS: System.TypeLoadException has been thrown with 9.0.3

### ISSUE THREAD

**Issue by inimirpaz** [CommunityToolkit/Maui#2148]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

Upgrading the `CommunityToolkit.Maui` package from 9.0.2 to 9.0.3 results in the app crashing at startup in iOS due to `System.TypeLoadException has been thrown`, `Could not set up parent class, due to: Invalid generic instantiation assembly:...Microsoft.Maui.dll type:ViewHandler`2 member:(null)`.

I have tried both with Maui 8.0.71 and 8.0.80.

One thing I noticed is that the dependencies of the NuGet have changed: 9.0.2 also required `Microsoft.Maui.Controls.Compatibility >= 8.0.21` but 9.0.3 does not.

### Expected Behavior

App not crashing, reverting back to 9.0.2 solves the issue.

### Steps To Reproduce

Open a new project, install 9.0.3, run on iOS.

### Link to public reproduction project repository

https://github.com/inimirpaz/maui_issues/tree/commtoolkit_ios_issue

### Environment

```markdown
- .NET MAUI CommunityToolkit: 9.0.3
- OS: MacOS
- .NET MAUI: 8.0.71 / 8.0.80
```

**Comment by RuddyOne** [comment]
I have also just upgraded and ran into this issue.

**Comment by danieltharris** [comment]
Also seeing this issue today. On a fresh MAUI project in Rider versioning MAUI to 8.0.71 works with Community Toolkit 9.0.2 for anybody who needs to get moving on a fresh project

**Comment by rdurfee** [comment]
I'm also replicating this issue.

**Comment by VladislavAntonyuk** [comment]
If you manually install Microsoft.Maui.Controls.Compatibility, does it solve the issue?

**Comment by brminnick** [comment]
I cannot reproduce this error using the sample provided.

I also cannot reproduce this error using `CommunityToolkit.Maui.Sample`.

This is likely a misconfiguration with your local environment. 

@inimirpaz Please ensure you have installed the following items:
- [ ] Xcode 15.4.0
- [ ] Update to the latest stable version of Visual Studio (or Jet Brains Rider)
- [ ] Download/install the [latest stable version of .NET 8 (v8.0.400)](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
- [ ] After installing .NET SDK v8.0.400, update to the latest stable version of .NET MAUI
  - On macOS, open the Terminal and enter the following command: `sudo dotnet workload install maui`
  - On Windows, open the command prompt (or Powershell) and enter the following command: `dotnet workload install maui`

**Comment by gautambjain** [comment]
Not getting this error after updating to Xcode 15.4.0 (earlier was 15.2.0). I had to also update to latest iOS simulators.

**Comment by inimirpaz** [comment]
> I cannot reproduce this error using the [provided reproduction sample](https://github.com/inimirpaz/maui_issues/tree/commtoolkit_ios_issue).
> 
> I also cannot reproduce this error using `CommunityToolkit.Maui.Sample`.
> 
> This is likely a misconfiguration with your local environment.
> 
> @inimirpaz Please ensure you have installed the following items:
> 
>     * [ ]  Xcode 15.4.0
> 
>     * [ ]  Update to the latest stable version of Visual Studio (or Jet Brains Rider)
> 
>     * [ ]  Download/install the [latest stable version of .NET 8 (v8.0.400)](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
> 
>     * [ ]  After installing .NET SDK v8.0.400, update to the latest stable version of .NET MAUI
>       
>       * On macOS, open the Terminal and enter the following command: `sudo dotnet workload install maui`
>       * On Windows, open the command prompt (or Powershell) and enter the following command: `dotnet workload install maui`

Thanks @brminnick the problem lay... the .NET SDK was not updated to the latest stable version (it was 8.0.300-something). I have updated it to 8.0.400 and the problem is gone.

**Comment by marcojak** [comment]
Same problem...
I have XCode 15.4
I've updated Rider to the latest version.
I've installed the latest stable version of .NET 8 (v8.0.401)
I've updated to the latest stable version of MAUI.

Still I get the same error.

**Comment by inimirpaz** [comment]
> Same problem... I have XCode 15.4 I've updated Rider to the latest version. I've installed the latest stable version of .NET 8 (v8.0.401) I've updated to the latest stable version of MAUI.
> 
> Still I get the same error.

If you run `dotnet workload list` what does it says?

**Comment by kevinjohnobrien** [comment]
Try running `dotnet restore` in the terminal.

**Comment by metalium84** [comment]
I'm still having this issue with .NET 8.0.402 and Xcode 15.4:
- CommunityToolkit.Maui 9.1.0
- Microsoft.Maui.* 8.0.91
```
Workload version: 8.0.402

Installed Workload Id      Installation Source
----------------------------------------------
maui-ios                   SDK 8.0.400        
ios                        SDK 8.0.400        
maui                       SDK 8.0.400
```

**Comment by admindeytabyte** [comment]
Having the same issue. Updated 

CommunityToolkit.Maui 9.1.0
Microsoft.Maui.* 8.0.91
Microsoft.Maui.Controls.Compatibility 8.0.82

System.TypeLoadException: 'Could not set up parent class, due to: Invalid generic instantiation assembly:/Users/name/Library/Developer/CoreSimulator/Devices/869D7B6E-ABA9-4743-9E23-A3D7B53E5B16/data/Containers/Bundle/Application/169D5C87-CF3D-4DEE-BAC3-5714FC7424FE/app/Microsoft.Maui.dll type:ViewHandler`2 member:(null)'

Even after updating the compatibility to 8.091 it reverts back to 8.0.82

**Comment by bijington** [comment]
Have you tried the steps in this comment https://github.com/CommunityToolkit/Maui/issues/2148#issuecomment-2308232294

**Comment by metalium84** [comment]
Yes, I did!
The only difference in my system is that I have installed both XCode 16.0 and 15.4 as I already updated to macOS Sequoia - using 15.4 Command Line Tools according to [this thread](https://dev.to/vhugogarcia/solving-net-maui-issues-on-macos-sequoia-with-visual-studio-code-and-xcode-16-3ljd).

**Comment by filipnavara** [comment]
Cross-ref: https://github.com/xamarin/xamarin-macios/issues/21335

TL;DR: This is reoccurring issue due to the way the iOS Workload bumps the implicitly bumps the SDK version. If you have .NET 9 RC 1 SDK installed on a machine then CommunityToolkit.Maui 9.1.0 will not work unless you force using .NET 8 SDK with global.json and update all workloads to latest version.

**Comment by bijington** [comment]
@filipnavara thank you for sharing this. It makes me happy to have a better understanding of what is happening. It sounds like we might need to opt into targeting specific versions like `net8.0-17.0`.

**Comment by XamMattia83** [comment]
I have the same issue. Working with Xcode 15.4 and Xcode 16 installed (command line tool set to 15.4) on MacOS Sonoma 14.6.1

On Windows side i have:
CommunityToolkit.Maui 9.1.0
Maui 8.0.91 
Maui sdk 8.0.4

result of workload list:

![image](https://github.com/user-attachments/assets/f712c81c-eca1-4708-8356-28b73f7ff0b6)

**Comment by XamMattia83** [comment]
> @XamMattia83 same issue here working with MacOS Sonoma 14.5, dotnet 8.0.402. Did you found any solution?
> 
> Workload version: 8.0.402.0
> 
> ## Installed Workload Id Installation Source
> ios SDK 8.0.400 maui SDK 8.0.400 android SDK 8.0.400

Nope, unfortunately ;( I'm still looking for it...

**Comment by shmoe01** [comment]
Why is this issue closed? this is still an issue for us. It also looks like others are struggling with this also? when will a new version be available that is compatible?

**Comment by bijington** [comment]
It was closed back in August when it was believed to be resolved. We find that most often this issue is caused by the developers environment. If you are experiencing the issue please try these steps https://github.com/CommunityToolkit/Maui/issues/2148#issuecomment-2308232294

**Comment by brminnick** [comment]
Yes, this is most likely a misconfiguration of your local environment. Please open a new Issue with a reproduction sample if you are unable to resolve the problem.

We cannot reproduce this error.

@shmoe01  Please ensure you have installed the following items:
- [ ] Xcode 16.0.0
- [ ] Update to the latest stable version of Visual Studio (or Jet Brains Rider)
- [ ] Download/install the [latest stable version of .NET 8 (v8.0.403)](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
- [ ] After installing .NET SDK v8.0.403, update to the latest stable version of .NET MAUI
  - On macOS, open the Terminal and enter the following command: `sudo dotnet workload install maui`
  - On Windows, open the command prompt (or Powershell) and enter the following command: `dotnet workload install maui`
